src/borg/helpers/fs.py

```diff
@@ -245,6 +245,8 @@ def remove_dotdot_prefixes(path):
 
     `path` is expected to be normalized already (e.g. via `os.path.normpath()`).
     """
+    if is_win32:
+        path = os.path.splitdrive(path)[1]
```

Comment:
what i already wondered: is borg already getting paths with slashes (and not backslashes)?

Reply:
added some logging, it looks like it is:
```
$ pytest -s -k "test_archived_paths and not remote and not binary"
=========================================================================================== test session starts ============================================================================================
platform win32 -- Python 3.10.9, pytest-7.2.0, pluggy-1.0.0
benchmark: 4.0.0 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
Tests enabled: root, hardlinks, modes
Tests disabled: BSD flags, fuse2, fuse3, symlinks, atime/mtime
rootdir: C:/msys64/home/Rayyan/borg, configfile: setup.cfg
plugins: benchmark-4.0.0, cov-4.0.0, forked-1.4.0, xdist-3.1.0
collected 1746 items / 1745 deselected / 1 selected

src/borg/testsuite/archiver/create_cmd.py
self.cmd output START
self.cmd output END


self.cmd output START
remove_dotdot_prefixes path =  input
remove_dotdot_prefixes path =  input/test
remove_dotdot_prefixes path =  C:/Users/Rayyan/AppData/Local/Temp/tmpz08pkzkq/input/test
self.cmd output END


self.cmd output START
input
input/test
Users/Rayyan/AppData/Local/Temp/tmpz08pkzkq/input/test
self.cmd output END


self.cmd output START
{"flags": 0, "gid": 0, "group": "root", "healthy": true, "hlid": "", "mode": "drwxrwxrwx", "mtime": "2023-06-25T21:38:16.244628+01:00", "path": "input", "size": 0, "target": "", "type": "d", "uid": 0, "user": "root"}
{"flags": 0, "gid": 0, "group": "root", "healthy": true, "hlid": "", "mode": "-rw-rw-rw-", "mtime": "2023-06-25T21:38:16.244628+01:00", "path": "input/test", "size": 0, "target": "", "type": "-", "uid": 0, "user": "root"}
{"flags": 0, "gid": 0, "group": "root", "healthy": true, "hlid": "", "mode": "-rw-rw-rw-", "mtime": "2023-06-25T21:38:16.244628+01:00", "path": "Users/Rayyan/AppData/Local/Temp/tmpz08pkzkq/input/test", "size": 0, "target": "", "type": "-", "uid": 0, "user": "root"}
self.cmd output END

.

==================================================================================== 1 passed, 1745 deselected in 1.63s ==================================================================================== 
```

Reply:
Do you know where exactly the usual backslashes on windows are transformed to slashes?

Reply:
just tested making an archive manually with borg.exe, looks like the paths are still backslashes...
strange. will look into this more.

Reply:
appears to work fine now but i don't know if posixpath.normpath is needed in other places as well